utils/lint_test_cases.py

```diff
@@ -0,0 +1,785 @@
+        )
+
+    def find_all_test_cases(self, pattern: str = "**/*.yaml") -> List[Path]:
+        """
+        Find all test case YAML files in the test-cases directory.
+
+        Args:
+            pattern: Glob pattern for finding files
+
+        Returns:
+            List of Path objects for test case files
+        """
+        return sorted(self.test_cases_dir.glob(pattern))
+
+    def group_by_backend_config(self, test_cases: Optional[List[Path]] = None) -> Dict[str, List[Path]]:
+        """
+        Group test cases by their backend configuration requirements.
+
+        Args:
+            test_cases: List of test case paths, or None to find all
+
+        Returns:
+            Dictionary mapping backend config keys to list of test case paths
+        """
+        if test_cases is None:
+            test_cases = self.find_all_test_cases()
+
+        grouped = {}
+
+        for test_case in test_cases:
+            analysis = self.analyze_file(test_case)
+            configs = self.get_backend_configs(analysis)
+
+            for config in configs:
+                key = self._config_key(config)
+                if key not in grouped:
+                    grouped[key] = []
+                grouped[key].append(test_case)
+
+        return grouped
+
+    def _config_key(self, config: BackendConfig) -> str:
+        """Generate a unique key for a backend configuration."""
+        parts = [config.backend.value]
+        if config.multi_region:
+            parts.append(f"regions-{config.region_count}")
+        return "-".join(parts)
+
+    def run_tests_for_file(
+        self,
+        file_path: Path,
+        verbose: bool = True,
+    ) -> Tuple[int, List[Dict[str, Any]]]:
+        """
+        Analyze a test case file and run appropriate YAML validation tests.
+
+        Args:
+            file_path: Path to test case file
+            verbose: If True, print detailed output
+
+        Returns:
+            Tuple of (total_exit_code, list of failure info dicts)
+        """
+        analysis = self.analyze_file(file_path)
+        configs = self.get_backend_configs(analysis)
+
+        total_exit_code = 0
+        failures = []
+
+        for config in configs:
+            exit_code, failure_info = self.run_yaml_test_for_config(config, file_path, verbose=verbose)
+            total_exit_code += exit_code
+            if failure_info:
+                failures.append(failure_info)
+
+        return total_exit_code, failures
+
+
+def _worker_process_file(file_path: Path) -> Tuple[int, List[Dict[str, Any]], Path]:
+    """
+    Worker function to process a single test file in a separate process.
+
+    This function is designed to be called by multiprocessing.Pool and must be
+    at module level (not a method) to be picklable.
+
+    Args:
+        file_path: Path to test case file
+
+    Returns:
+        Tuple of (exit_code, failures, file_path)
+    """
+    try:
+        analyzer = TestCaseAnalyzer()
```

Comment:
We are initialiazing class for both process and analyze, this tell me that the class has too many responsibilities or we can share the class among the functions.